uxarray/grid.py

```diff
@@ -379,6 +384,100 @@ def integrate(self, var_ds, quadrature_rule="triangular", order=4):
+        mesh2_edge_nodes[:, 0] = pad_results[:, :-1].ravel()
+        mesh2_edge_nodes[:, 1] = pad_results[:, 1:].ravel()
+
+        # Clean up the invalid edge (same node to same node except the [-1, -1] edge)
+        valid_mask = np.array(
+            list(
+                map(
+                    lambda edge: not (edge[0] == edge[1] and not np.array_equal(
+                        edge, np.array([-1, -1]))), mesh2_edge_nodes)))
+        mesh2_edge_nodes = mesh2_edge_nodes[valid_mask]
+
+        # # Find the unique edge
+        mesh2_edge_nodes.sort(axis=1)
+        mesh2_edge_node_copy, inverse_indices = np.unique(ar=mesh2_edge_nodes,
+                                                          return_inverse=True,
+                                                          axis=0)
+
+        # In mesh2_edge_nodes, we want to remove all dummy edges (edge that has "-1" node index)
+        # But we want to preserve that in our mesh2_face_edges so make the datarray has the same dimensions
+        cumprod = (mesh2_edge_node_copy[:, 0] + 1) * \
+                  (mesh2_edge_node_copy[:, 1] + 1) # In this case, former 0 will be positive while former -1 will be 0
+        has_fill_value = np.ma.masked_where(cumprod != 0, cumprod).mask
+        mesh2_edge_nodes = mesh2_edge_node_copy[has_fill_value] if isinstance(
+            has_fill_value, np.ndarray) else mesh2_edge_node_copy
+        inverse_indices = inverse_indices.reshape(n, m)
+        mesh2_face_edges = mesh2_edge_node_copy[inverse_indices]
+
+        # TODO: Reorder each edge such that they're in the counterclockwise
+        # First we need to recover the original order from the `Mesh2_face_nodes`. Since we know `Mesh2_face_nodes` are
+        # stored in the counterclockwise order, we just need to construct the first edge to predict the following.
+        mesh2_face_edges[:, 0] = np.stack(
+            (mesh2_face_nodes[:, 0], mesh2_face_nodes[:, 1]), axis=-1)
+        for i in range(0, mesh2_face_nodes):
+            # We need to make sure cur_edge[0] has the same node index as the prev_edge[1]
+            for j in range(1, len(mesh2_face_nodes[i])):
+                last_node = mesh2_face_nodes[i][j - 1][1]
+                match_index_cur_edge = np.where(
+                    mesh2_face_nodes[i][j] == last_node)
+                match_index_cur_edge = match_index_cur_edge[
+                    0] if match_index_cur_edge[0].size else -1
+                if match_index_cur_edge == 0:
+                    continue
+                elif match_index_cur_edge == 1:
+                    # Now switch two indexes
+                    temp = mesh2_face_nodes[i][j][0]
+                    mesh2_face_nodes[i][j][0] = last_node
+                    mesh2_face_nodes[i][j][1] = temp
+
+        self.ds["Mesh2_face_edges"] = xr.DataArray(
+            data=mesh2_face_edges,
+            dims=["nMesh2_face", "nMaxMesh2_face_edges", "Two"],
```

Comment:
```suggestion
            dims=["nMesh2_face", "nMaxMesh2_face_edges"],
```

From the UGRID conventions:
```
integer Mesh2_face_edges(nMesh2_face, nMaxMesh2_face_nodes) ;